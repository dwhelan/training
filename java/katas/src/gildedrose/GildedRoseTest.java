package gildedrose;

import org.approvaltests.Approvals;
import org.approvaltests.reporters.DiffReporter;
import org.approvaltests.reporters.UseReporter;
import org.junit.Test;

public class GildedRoseTest {

	@Test
	@UseReporter(DiffReporter.class)	
	public void test() throws Exception {
		Item[] items = new Item[] {
				new Item("+5 Dexterity Vest", 10, 20), 
				new Item("Aged Brie", 2, 50),
				new Item("Aged Brie", 0, 50),
				new Item("Aged Brie", -1, 5),
				new Item("Backstage passes to a TAFKAL80ETC concert",  5, 20),
				new Item("Backstage passes to a TAFKAL80ETC concert",  5, 49),
				new Item("Backstage passes to a TAFKAL80ETC concert", 10, 20),
				new Item("Backstage passes to a TAFKAL80ETC concert", 10, 49),
				new Item("Backstage passes to a TAFKAL80ETC concert", 11, 20),
				new Item("Backstage passes to a TAFKAL80ETC concert",  0, 20),
				new Item("Sulfuras, Hand of Ragnaros", 0, 1),
				new Item("Sulfuras, Hand of Ragnaros", -1, 1),
				new Item("Elixir of the Mongoose", 0, 0),
				new Item("Elixir of the Mongoose", 0, 7),
		};

		new GildedRose(items).updateQuality();

		Approvals.verifyAll("Items", items);
	}
}
