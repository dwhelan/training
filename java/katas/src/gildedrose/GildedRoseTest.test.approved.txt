Items[00] = +5 Dexterity Vest, 9, 19
Items[01] = Aged Brie, 1, 50
Items[02] = Aged Brie, -1, 50
Items[03] = Aged Brie, -2, 7
Items[04] = Backstage passes to a TAFKAL80ETC concert, 4, 23
Items[05] = Backstage passes to a TAFKAL80ETC concert, 4, 50
Items[06] = Backstage passes to a TAFKAL80ETC concert, 9, 22
Items[07] = Backstage passes to a TAFKAL80ETC concert, 9, 50
Items[08] = Backstage passes to a TAFKAL80ETC concert, 10, 21
Items[09] = Backstage passes to a TAFKAL80ETC concert, -1, 0
Items[10] = Sulfuras, Hand of Ragnaros, 0, 1
Items[11] = Sulfuras, Hand of Ragnaros, -1, 1
Items[12] = Elixir of the Mongoose, -1, 0
Items[13] = Elixir of the Mongoose, -1, 5
